src/base/rss/rss_feed.cpp

```diff
@@ -424,6 +410,86 @@ void Feed::downloadIcon()
             , this, &Feed::handleIconDownloadFinished);
 }
 
+int Feed::updateArticles(QList<QVariantHash> loadedArticles)
+{
+    if (loadedArticles.count() == 0)
+        return 0;
+
+    QDateTime dummyPubDate {QDateTime::currentDateTime()};
+    QVector<QVariantHash> newArticles;
+    newArticles.reserve(loadedArticles.size());
+    for (QVariantHash article : loadedArticles) {
+        QVariant &torrentURL = article[Article::KeyTorrentURL];
+        if (torrentURL.toString().isEmpty())
+            torrentURL = article[Article::KeyLink];
+
+        // if item does not have a guid, fall back to some other identifier
+        QVariant &localId = article[Article::KeyId];
+        if (localId.toString().isEmpty())
+            localId = article.value(Article::KeyTorrentURL);
+        if (localId.toString().isEmpty())
+            localId = article.value(Article::KeyTitle);
+
+        if (localId.toString().isEmpty())
+            continue;
+
+        // If article has no publication date we use feed update time as a fallback.
+        // To prevent processing of "out-of-limit" articles we must not assign dates
+        // that are earlier than the dates of existing articles.
+        const Article *existingArticle = articleByGUID(localId.toString());
+        if (existingArticle) {
+            dummyPubDate = existingArticle->date().addMSecs(-1);
+            continue;
+        }
+
+        QVariant &articleDate = article[Article::KeyDate];
+        // if article has no publication date we use feed update time as a fallback
```

Comment:
@glassez:  **Grammar Alert:**

>**If** article has no publication date we use feed update time as a **fallback.**

Likewise.  Also note that this sentence is repeated in the preceding code-comment—but with proper capitalization and punctuation.